Recruit-Training/Advanced-Recruit-Training/Embedded-Teaser/Training/Core/Src/Training.c

```diff
@@ -2,5 +2,59 @@
 
 void trainingTask(uint8_t* data)
 {
+	validData = 0;
+	
+	if (motorCheck(*data) && motorCheck(*data) == motorCheck(*(data + 1)) && lightsCheck(*(data + 2)) ) // set outputArray if motors data are valid, motors are in sync and lights data is valid
+	{
+		for (int i = 0; i < sizeof outputArray / sizeof *outputArray; i++)
+		{
+			outputArray[i] = *(data + i);
+		}	
+	}
+	else
+	{
+		for (int i = 0; i < sizeof outputArray / sizeof *outputArray; i++)
+		{
+			outputArray[i] = 0;
+		}	
+	}
+	
+	if (motorCheck (*data))  // set first motor bit in validData to 1 if motor's data is valid
+	{
+		validData = validData | (1 << FIRSTMOTOR);
+	}
+	
+	if (motorCheck (*(data + 1))) // set second motor bit in validData to 1 if motor's data is valid
+	{
+		validData = validData | (1 << SECONDMOTOR);
+	}
+	if (lightsCheck (*(data+2))) // set lights bit in validData to 1 if lights data is valid
+	{
+		validData = validData | (1 << LIGHTSBOARD);
+	}	
+}
+
+int motorCheck (uint8_t testMotor)
+{
+	uint8_t motorMask = 0b11000001;
+	
+	if ((testMotor & motorMask) == 0b10000000) // check the case that motor is in reverse (power on, negative velocity and reverse)
```

Comment:
If the sign of the velocity is 1, the velocity is negative, if the sign of the velocity is 0, the velocity is positive